src/main/java/org/wildfly/security/http/impl/BasicAuthenticationMechanism.java

```diff
@@ -183,10 +192,18 @@ public void evaluateRequest(final HttpServerRequest request) throws HttpAuthenti
             }
         }
 
-        request.noAuthenticationInProgress(response -> prepareResponse(displayRealmName, response));
+        request.noAuthenticationInProgress(response -> prepareResponse(request, displayRealmName, response));
     }
 
-    private void prepareResponse(String realmName, HttpServerResponse response) {
+    private void prepareResponse(final HttpServerRequest request, String realmName, HttpServerResponse response) {
+        if (silent) {
+            //if silent we only send a challenge if the request contained auth headers
+            //otherwise we assume another method will send the challenge
+            String authHeader = request.getFirstRequestHeaderValue(AUTHORIZATION);
+            if(authHeader == null) {
```

Comment:
Shouldn't be checked here that Authorization header is of Basic mechanism? Can it happen this mechanism can see  Digest mechanism header?

Reply:
I think if there is Digest mechanism header, it might have successful credentials and so it is better to not send challenge in this case also.